### ISSUE TITLE
Add SMS Support Virgin Mobile Canada

### ISSUE THREAD

**Issue by Francewhoa** [webmin/webmin#1187]
Hello Webmin enthusiasts :)

This is a suggestion to add support for SMS with **Virgin Mobile Canada**

- SMTP Format: `<10digitphonenumber>@vmobile.ca`
- Supplier name: `Virgin Mobile Canada`

At Ubertus we would be happy to contribute testing and documentation if needed

Assumptions
- Assumes the present Webmin version 1.941 support Virgin Mobile USA. But not Virgin Mobile Canada. 
- Assumes the challenge is that Virgin Mobile USA does not work for Virgin Mobile Canada. As their SMTP to SMS gateway are different. Specifically Virgin Mobile USA SMTP format are either`<phonenumber>@vmobl.com` or `<phonenumber>@vxtras.com`. Compare to Virgin Mobile Canada `<10digitphonenumber>@vmobile.ca`
- Assumes this is the source of the SMTP Format for Virgin Mobile Canada https://www.notepage.net/smtp.htm
- Assumes this suggestion is about this `Send SMS to` field in this screenshot:
![Add SMS Support Virgin Mobile Canada](https://user-images.githubusercontent.com/119942/73126809-e5f04200-3f6b-11ea-9a86-1d3a6ae5748a.jpg)
- Assumes related discussion at https://forum.virtualmin.com/t/mobile/27039

**Comment by chris001** [comment]
Webmin's list of email to SMS gateway addresses seem to be only a small fraction of the available gateways right now.

Here are seven updated addresses, which could/should all be added to webmin, to support nearly all mobile providers of the world:

1. https://kb.sandisk.com/app/answers/detail/a_id/17056/~/list-of-mobile-carrier-gateway-addresses
2. https://support.teamunify.com/en/articles/227
3. https://raw.githubusercontent.com/mfitzp/List_of_SMS_gateways/master/email2sms.csv
4. https://en.wikipedia.org/wiki/SMS_gateway
5. https://www.opentextingonline.com/emailtotext.aspx
6. https://smsemailgateway.com/
7. https://avtech.com/articles/138/list-of-email-to-sms-addresses/

**Comment by Francewhoa** [comment]
@chris001 :) I agree it would be nice for Webmin to support all SMS gateway addresses worldwide. I suggest to create a new ticket for this new request. As you know there are a lot SMS gateway addresses worldwide. So this might be a significant amount of work to add, test, document, all of those. Any volunteer to contribute a patch for this? I suggest to start with the most popular SMS gateway addresses first.

As for this ticket #1187 my suggestion includes and is limited to adding support for Virgin Mobile Canada. Which is part of the top 4 largest mobile network operators in Canada. All together they represent 82%+ of all SMS gateway addresses in Canada. Source:
- https://en.wikipedia.org/wiki/List_of_Canadian_mobile_phone_companies#Mobile_network_operators
- https://en.wikipedia.org/wiki/Virgin_Mobile_Canada

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.

**Comment by swelljoe** [comment]
As Virgin Mobile Canada is now Virgin Plus, and as I can't find any evidence they have an email-to-text gateway, I've closed my PR that would have added it. If someone with Virgin Plus wants to confirm there is an email-to-test gateway and if anyone is still actually using email-to-text notifications rather than just getting email, I'll add it.

But, I'm inclined to say the time of email-to-text gateways has passed. We all have had email on phones for two decades.

**Comment by jcameron** [comment]
> But, I'm inclined to say the time of email-to-text gateways has passed. We all have had email on phones for two decades.

Indeed! That said, maybe in future we could look at integrating with something like Pagerduty

**Comment by Francewhoa** [comment]
> As Virgin Mobile Canada is now Virgin Plus, and as I can't find any evidence they have an email-to-text gateway,

Hi @swelljoe, the new Virgin Plus' email-to-text address format is:
`<10 DIGITS PHONE NUMBER>@txt.virgin.ca`

Example:
`1234567890@txt.virgin.ca`

According to Virgin Mobile Canada, it is normal that the email address is either not published or hard to find. They happily give it to their customers though. I am assuming they do this to reduce spam.

**Comment by Francewhoa** [comment]
>But, I'm inclined to say the time of email-to-text gateways has passed. We all have had email on phones for two decades.

Hi @swelljoe and @swelljoe, I hople that both you and all your loved ones are happy and healthy.

For your information and for your consideration. I agree that some people don't need email-to-text. While at the same time, others do. To each their unique needs. 

I do still frequently need email-to-text. Because email-to-text is very useful for rural and remote area. Specifically when forwarded to SMS teams' text messages. For example, while travelling or hiking in the wilderness, the challenge with Wi-Fi is that it is either not available or spotty at best. In comparison, SMS has a much longer range. 

The other major benefit with SMS sent via email-to-text is that as soon as there is the smallest of connection, the SMS are received. Even if they were sent hours or days ago. For example, when hiking up and down 7,500 feet mountains areas, little or no SMS are receiving at low elevation. But at each summit, must SMS are fully received. Joy :) In comparison, no Wi-Fi or phone connection at neither low elevation or summits :( In case of a real emergency, it is very useful to be able to both receive and send SMS from the jungle ;)

A frequent case is when a senior IT person is enjoying a trip to the wilderness, but the junior team has a real emergency, when appropriate and agreed on, they can contact her/him via SMS. On top of that, many emergency workers, such as wildfire firefighter use SMS for backup communications. Which saved countless lives. Both human and animal. And properties.
